packages/manifold/contracts/gachaclaims/IGachaLazyClaim.sol

```diff
@@ -0,0 +1,105 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.0;
+
+/// @author: manifold.xyz
+
+/**
+ * Gacha Lazy Claim interface
+ */
+interface IGachaLazyClaim {
+  enum StorageProtocol {
+    INVALID,
+    NONE,
+    ARWEAVE,
+    IPFS
+  }
+
+  error InvalidStorageProtocol();
+  error InvalidDate();
+  error InvalidInstance();
+  error InvalidInput();
+  error InvalidPayment();
+  error InvalidSignature();
+  error InvalidVariationIndex();
+  error InvalidStartingTokenId();
+  error ClaimAlreadyInitialized();
+  error ClaimNotInitialized();
+  error ClaimInactive();
+  error ClaimSoldOut();
+  error TokenDNE();
+  error FailedToTransfer();
+  error TooManyRequested();
+  error CannotLowerTotalMaxBeyondTotal();
+  error CannotChangeTokenVariations();
+  error CannotChangePaymentToken();
+  error CannotLowertokenVariationsBeyondVariations();
+  error CannotMintMoreThanReserved();
+  error CannotMintFromContract();
+
+  event GachaClaimInitialized(address indexed creatorContract, uint256 indexed instanceId, address initializer);
+  event GachaClaimUpdated(address indexed creatorContract, uint256 indexed instanceId);
+  event GachaClaimMintReserved(
+    address indexed creatorContract,
+    uint256 indexed instanceId,
+    address indexed collector,
+    uint32 mintCount
+  );
+
+  struct Recipient {
+    uint256 mintCount;
+    address receiver;
+  }
+
```

Comment:
We need a better structure so we aren't repeatedly looping and retrieving a claim unecessarily.
The structure should really be:
struct ClaimMint {
  address creatorContractAddress;
  uint256 instanceId;
  VariationMint[] variationMint;
}

struct VariationMint {
  uint8 variationIndex;
  Recipient[] recipients;
}

This way you can do one outer for loop and retrieve the Claim only once, and do one variation mint per variation to all recipients.

Reply:
Wait am I missing something or will this require 3 loops? Are we no longer accepting an array of mints / claimMints?

Agree that this will save on loops required to call mint function but we would still need to loop through all the recipients to check against reserved and update delivered right?

Reply:
and actually when i was fixing the references i found some bugs with the order of how things were getting incremented which might contributing to above ^^

Reply:
Yes. You still need to loop recipients to update counts.

Reply:
Wait, why not? I can think of how the cron can easily figure this out in the mint script.  @jaxonL

Reply:
@nikki-kiga @jaxonL just changed it.  Don't need Recipient struct.

Why?  Because the external mint call can mint multiple tokens to multiple receivers in multiple amounts in one shot.